apps/versions/views.py

```diff
@@ -110,12 +110,16 @@ def guard():
 
 
 @addon_view_factory(guard)
-def download_latest(request, addon, type='xpi', platform=None):
+def download_latest(request, addon, beta=False, type='xpi', platform=None):
     platforms = [amo.PLATFORM_ALL.id]
     if platform is not None and int(platform) in amo.PLATFORMS:
         platforms.append(int(platform))
+    if beta and addon.show_beta:
```

Comment:
Yes, currently we would server a stable release. I was wondering whether providing something would be better than giving nothing in that case. Then again, it would probably be unexpected behavior - and it's not like the beta channel would ever go away once it is present. So I guess 404 would indeed be better.